### ISSUE TITLE
Could not find device for node: {{node FloorMod}} = FloorMod[T=DT_HALF]

### ISSUE THREAD

**Issue by shkarupa-alex** [tensorflow/tensorflow#45920]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Google Colab
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): Google Colab
- TensorFlow version (use command below): v2.4.0-0-g582c8d236cb 2.4.0
- Python version: 3
- Bazel version (if compiling from source): No
- GCC/Compiler version (if compiling from source): No
- CUDA/cuDNN version: No
- GPU model and memory: No

**Describe the current behavior**
Since TF 2.4.0 i get an error during FloorMod operation with float16.
In TF 2.3.1 it worked as expected.

**Describe the expected behavior**
FloorMod should be available for float16 dtype.

**Standalone code to reproduce the issue**
https://colab.research.google.com/drive/19kRHgXjuf9R9UuBYmEmZiFQPpodnbsOM?usp=sharing

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
```python
import tensorflow as tf

print(tf.version.GIT_VERSION, tf.version.VERSION)
# v2.4.0-0-g582c8d236cb 2.4.0

x = tf.constant([1, 2, 3], 'float16')
x % 4

---------------------------------------------------------------------------
NotFoundError                             Traceback (most recent call last)
<ipython-input-6-86e5a110f54d> in <module>()
----> 1 x % 4

3 frames
/usr/local/lib/python3.6/dist-packages/six.py in raise_from(value, from_value)

NotFoundError: Could not find device for node: {{node FloorMod}} = FloorMod[T=DT_HALF]
All kernels registered for op FloorMod:
  device='GPU'; T in [DT_INT32]
  device='CPU'; T in [DT_DOUBLE]
  device='CPU'; T in [DT_FLOAT]
  device='CPU'; T in [DT_BFLOAT16]
  device='CPU'; T in [DT_UINT64]
  device='CPU'; T in [DT_INT64]
  device='CPU'; T in [DT_INT32]
 [Op:FloorMod]
```

**Comment by ravikyram** [comment]
I have tried in colab with TF 2.4 , nightly version(`2.5.0-dev20201222`) and was able to reproduce the issue. However I am not seeing any issue with TF 2.3 version. Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/23c82e886865b955bc4f9f36058ba948/untitled568.ipynb). Thanks!

**Comment by shkarupa-alex** [comment]
Still not fixed in v1.12.1-52612-g74d34665fc1 2.5.0-dev20210311

**Comment by shkarupa-alex** [comment]
Fixed in v1.12.1-54850-g958d4f19097 2.6.0-dev20210414
